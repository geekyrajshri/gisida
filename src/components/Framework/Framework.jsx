import React from 'react';
import PropTypes from 'prop-types';

require('./Framework.scss');

class Framework extends React.Component {
  componentDidMount() {
    this.getKey();
  }
  getFrameworkSectors() {
    const sectors = this.props.sectorData.filter(sector => sector.sectors).map(sector => sector);
    const frameworkSector = [];
    sectors.forEach(sector =>
      frameworkSector.push((<div className="framework-sector">
        <div className="sector-header">
          <div className="header">{sector.sector}</div>
          <img src={sector.icon} alt={sector.sector} className="icon" />
        </div>
        <ul>
          {sector.sectors.map(layer =>
            (
              <li className="layerItem" key={layer}>
                {sector.headers.includes(layer) ?
                  <div className="sub-sector">{layer}</div> :
                  <div>
                    <div className="status">
                      <div className="status-1" />
                      <div className="status-2" />
                    </div>
                    <div className="status-link">
                      <a onClick={this.props.onToggleView}>{layer}</a>
                    </div>
                  </div>
                }
              </li>
            ))
          }
        </ul>
      </div>),
      ));
    return frameworkSector;
  }

  getKey() {
    const colors = ['white', 'transparent', 'red', 'orange', 'green'];
    let status = '';
    colors.forEach((color) => {
      const border = color === 'white' ? '1px solid #eee' : color === 'transparent' ? '1px dotted #555' : '';
      status += `<li style="background: ${color}; border: ${border};"></li>`;
    });
    $('.key').prepend(`<ul><li id="key-label">Key</li> ${status} </ul>`);
  }

  render() {
    return (
      <div className="framework-wrapper">
        <div className="filter-selection">
          <div>Your Selections: </div>
          <div className="selections">IDPs, BAIDOA - SOMALIA, 2017</div>
        </div>
        <div className="key" />
        <div className="framework-sectors">
          {this.getFrameworkSectors()}
        </div>
        <div className="powered-by" />
      </div>
    );
  }
}

Framework.propTypes = {
  sectorData: PropTypes.arrayOf(PropTypes.any).isRequired,
  onToggleView: PropTypes.func.isRequired,
};

export default Framework;
