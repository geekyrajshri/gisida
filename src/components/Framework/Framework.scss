    .framework-wrapper {
        position: relative;
        height: 100vh;
        .key {
            top: 10%;
            right: 15px;
            position: fixed;
            ul {
                li {
                    float: left;
                    #key-label {
                        width: 30px;
                        font-size: 14px;
                    }
                    .status-key {
                        width: 15px;
                        height: 15px;
                    }
                }
            }
        }
        .key-popup {
            display: none;
        }
        .key:hover+.key-popup {
            display: block;
        }
        .key-popup {
            z-index: 1001;
            position: fixed;
            top: 14%;
            right: 15px;
            width: 20%;
            background: #FFF;
            height: 25%;
            border: 1px #eee solid;
            box-shadow: 0 1px 7px rgba(0, 0, 0, 0.2);
            #popup-key-label {
                font-size: 14px;
                font-weight: 600;
                margin: 10px;
            }
            div {
                display: table;
                padding-left: 10px;
                padding-bottom: 2px;
                .popup-status-key {
                    display: table-cell;
                    width: 15px;
                    height: 15px;
                }
                .popup-status-description {
                    display: table-cell;
                    font-size: 12px;
                }
            }
        }
        .framework-sectors {
            position: fixed;
            top: 8.5em;
            width: 100%;
            margin-left: 2%;
            left: 16%;
            z-index: 1000;
            color: #555;
            font-size: 12px;
            line-height: 1.2;
            overflow-y: scroll;
            .framework-sector {
                width: 25%;
                margin-left: 2%;
                float: left;
                padding: 15px;
                border: 1px #eee solid;
                box-shadow: 0 1px 7px rgba(0, 0, 0, 0.2);
                .sector-header {
                    height: 45px;
                    border-bottom: 1px solid #eee;
                    .header {
                        float: left;
                        font-size: 22px;
                        font-weight: bold;
                    }
                    .icon {
                        float: right;
                        width: 30px;
                        text-align: center;
                        color: #ccc;
                        background: no-repeat center center;
                        background-size: 70%;
                    }
                }
                ul {
                    list-style: none;
                    margin-left: 4%;
                    li {
                        margin-left: -15%;
                        padding-top: 1.3%;
                        .sub-sector {
                            font-weight: bold;
                            padding-top: 3%;
                        }
                        div {
                            .sector-indicator {
                                display: table;
                                .status {
                                    display: -webkit-inline-box;
                                    border: 1px solid black;
                                    width: 25px;
                                    height: 25px;
                                    .status-1 {
                                        width: 50%;
                                        height: 100%;
                                    }
                                    .status-2 {
                                        width: 50%;
                                        height: 100%;
                                    }
                                }
                                .rectangle {
                                    width: 10%;
                                    height: 10%;
                                }
                                .status-link {
                                    cursor: pointer;
                                    display: table-cell;
                                    vertical-align: middle;
                                    padding-left: 10px;
                                }
                                a {
                                    color: #555;
                                    text-decoration: none;
                                }
                            }
                        }
                    }
                }
            }
        }
        .powered-by {
            width: 100px;
            position: absolute;
            bottom: 15px;
            right: 15px;
            content: url("/assets/img/Ona-powered-2color-horiz.png");
        }
    }

    .toggle-view-link {
        margin: 5px;
        display: inline-table;
    }

    .modal-body {
        h6 {
            font-family: inherit;
            font-weight: 600;
            line-height: 1.1;
            color: inherit;
        }
    }