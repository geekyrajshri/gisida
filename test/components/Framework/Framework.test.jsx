import React from 'react';
import renderer from 'react-test-renderer';
import jQuery from 'jquery';
import Framework from '../../../src/components/Framework/Framework';
import sectorData from '../../fixtures/sectors.json';
import layerData from '../../fixtures/layers.json';

window.$ = jQuery;

describe('Framework', () => {
  const onToggleView = jest.fn();
  const onViewClick = jest.fn();
  const onSectorClick = jest.fn();
  it('Framework renders each Framework view', () => {
    const component = renderer.create(
      <Framework
        onToggleView={onToggleView}
        onViewClick={onViewClick}
        onSectorClick={onSectorClick}
        layerData={[layerData]}
        sectorData={[sectorData]}
/>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
