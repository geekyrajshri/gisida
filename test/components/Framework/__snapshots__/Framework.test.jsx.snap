// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Framework Framework renders each Framework view 1`] = `
<div
  className="framework-wrapper"
>
  <div
    className="key"
  />
  <div
    className="key-popup"
  />
  <div
    className="framework-sectors"
  />
  <div
    className="footer"
  >
    <div
      className="powered-by"
    />
  </div>
</div>
`;
